jubilant/_juju.py

```diff
@@ -100,45 +110,83 @@ def deploy(
 
         self.cli(*args)
 
-    def status(
-        self,
-        *,
-        model: str | None = None,
-    ) -> Status:
+    def status(self) -> Status:
         """TODO."""
         args = ['status', '--format', 'json']
-
-        if model is None:
-            model = self.model
-        if model is not None:
-            args.extend(['--model', model])
+        if self.model is not None:
+            args.extend(['--model', self.model])
 
         stdout = self.cli(*args)
         result = json.loads(stdout)
         return Status.from_dict(result)
 
-    def wait_status(
+    def wait(
         self,
-        ready_func: typing.Callable[[Status], bool],
+        ready: Callable[[Status], bool],
         *,
-        model: str | None = None,
-        timeout: float = 10 * 60,
-        delay: float = 1,
+        error: Callable[[Status], bool] | None = any_error,  # TODO: default to None?
+        delay: float = 1.0,
+        timeout: float | None = None,
         successes: int = 3,
```

Comment:
suppose I start with delay=10. So the status has to keep stable for 30 seconds before it exits.
Now I want to drop the delay to 2 seconds to win some runtime, but I have to remember to bump successes to 15 to keep that part steady. That's not obvious.

Reply:
I was thinking the other way: if "soak" is in seconds, and you start with `delay=1, soak=3`, then you bump up to `delay=2` seconds, suddenly you're only requiring 1 successful call, which seems bad. I don't think there's a clearly right approach here, but having it as a number seems more straight-forward to me. Others?

Reply:
I think in theory this could go either way (per your example or mine). However, having the integer number rather than the time at least means you'll get multiple statuses having to return success. In addition, this is similar to other APIs like Kubernetes probes with [`successThreshold`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes) being a number of successes. So I think I'm going to leave this as is.